utils/docker/Dockerfile.centos.8

```diff
@@ -220,7 +220,16 @@ ARG DAOS_BUILD=$DAOS_DEPS_BUILD
 
 # Build DAOS
 RUN [ "$DAOS_BUILD" != "yes" ] || \
-    { scons-3 --jobs $JOBS install PREFIX=/opt/daos COMPILER=$COMPILER && \
+    { scons-3 --jobs $JOBS client install PREFIX=/opt/daos COMPILER=$COMPILER && \
+    utils/check.sh /opt/daos/bin/dmg && \
+    utils/check.sh -n /opt/daos/bin/daos_engine; }
+RUN [ "$DAOS_BUILD" != "yes" ] || \
+    { scons-3 --jobs $JOBS server install PREFIX=/opt/daos COMPILER=$COMPILER && \
+    utils/check.sh /opt/daos/bin/daos_engine && \
+    utils/check.sh -n /opt/daos/bin/vos_tests; }
+RUN [ "$DAOS_BUILD" != "yes" ] || \
+    { scons-3 --jobs $JOBS test install PREFIX=/opt/daos COMPILER=$COMPILER && \
+    utils/check.sh /opt/daos/bin/vos_tests && \
```

Comment:
I don't like the idea of test code in the dockerfiles, and we've been trying hard to harmonise the dockerfiles across distros as well, this logic would be much better in the utils/run_in_ga.sh script where it'll actually get run more often as well, and have better ways of accessing the output.

Reply:
Speaking of run_in_ga.sh, was it your intention to run NLT on the debug build? The last scons install there will do that.

Reply:
Actually, I can see the utility in case you use fault injection in NLT.   But why not use dev build?   Building with debug means -Od vs -O2 while both enable fault injection.

Reply:
I was previously running fault injection in github-actions but I backed it out as it takes so long and was less reliable than Jenkins.  I don't have a strong preference for what type of build we use for NLT here, they should all work.

One thing we've discovered recently however is we have poor build coverage of both dev and debug builds as there have been a few fixes landed recently.  https://github.com/daos-stack/daos/pull/5909 improves ga somewhat, the next step will be to add the missing matrix elements.